pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -141,50 +149,65 @@ func clusterFromProject(project string) (string, string) {
 	case oldPRTB.UserName != newPRTB.UserName && oldPRTB.UserName != "":
-		invalidFieldName = "userName"
+		return field.Invalid(fieldPath.Child("userName"), newPRTB.UserName, reason)
 	case oldPRTB.UserPrincipalName != newPRTB.UserPrincipalName && oldPRTB.UserPrincipalName != "":
-		invalidFieldName = "userPrincipalName"
+		return field.Invalid(fieldPath.Child("userPrincipalName"), newPRTB.UserPrincipalName, reason)
 	case oldPRTB.GroupName != newPRTB.GroupName && oldPRTB.GroupName != "":
-		invalidFieldName = "groupName"
+		return field.Invalid(fieldPath.Child("groupName"), newPRTB.GroupName, reason)
 	case oldPRTB.GroupPrincipalName != newPRTB.GroupPrincipalName && oldPRTB.GroupPrincipalName != "":
-		invalidFieldName = "groupPrincipalName"
+		return field.Invalid(fieldPath.Child("groupPrincipalName"), newPRTB.GroupPrincipalName, reason)
 	case (newPRTB.GroupName != "" || oldPRTB.GroupPrincipalName != "") && (newPRTB.UserName != "" || oldPRTB.UserPrincipalName != ""):
-		invalidFieldName = "both user and group"
+		return field.Forbidden(fieldPath,
+			"binding must target either a user [userName]/[userPrincipalName] OR a group [groupName]/[groupPrincipalName]")
+	case newPRTB.ServiceAccount != "":
+		return field.Forbidden(fieldPath.Child("serviceAccount"), "update is not allowed")
 	default:
 		return nil
 	}
-
-	return fmt.Errorf("cannot update %s for clusterRoleTemplateBinding %s: %w", invalidFieldName, oldPRTB.Name, admission.ErrInvalidRequest)
 }
 
 // validateCreateFields checks if all required fields are present and valid.
-func (v *Validator) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBinding) error {
+func (v *Validator) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBinding, fieldPath *field.Path) error {
+	if newPRTB.ServiceAccount != "" {
+		return field.Forbidden(fieldPath.Child("serviceAccount"), "")
+	}
+
 	hasUserTarget := newPRTB.UserName != "" || newPRTB.UserPrincipalName != ""
 	hasGroupTarget := newPRTB.GroupName != "" || newPRTB.GroupPrincipalName != ""
 
 	if (hasUserTarget && hasGroupTarget) || (!hasUserTarget && !hasGroupTarget) {
-		return fmt.Errorf("binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]: %w", admission.ErrInvalidRequest)
+		return field.Forbidden(fieldPath,
+			"binding must target either a user [userName]/[userPrincipalName] OR a group [groupName]/[groupPrincipalName]")
 	}
 
 	if newPRTB.ProjectName == "" {
-		return fmt.Errorf("binding must have field projectName set: %w", admission.ErrInvalidRequest)
+		return field.Required(fieldPath.Child("projectName"), "")
+	}
+
+	if newPRTB.RoleTemplateName == "" {
+		return field.Required(fieldPath.Child("roleTemplateName"), "")
 	}
 
 	roleTemplate, err := v.roleTemplateResolver.RoleTemplateCache().Get(newPRTB.RoleTemplateName)
 	if err != nil {
-		return fmt.Errorf("unknown reference roleTemplate '%s': %w", newPRTB.RoleTemplateName, err)
+		return err
```

Comment:
I don't follow why this can't also be a `fieldErr` returned? Something like:
```
return field.InternalError(fieldPath.Child("roleTemplate"), err)
```

That way we can avoid the `errors.As` check above.